contracts/RenPool.sol

```diff
@@ -70,16 +77,44 @@ contract RenPool {
         uint senderBalance = balances[sender];
 
         require(senderBalance > 0 && senderBalance >= _amount, "Insufficient funds");
-        require(isLocked == false, "Pool is locked");
+        if(!isLocked){
+            totalPooled -= _amount;
+            balances[sender] -= _amount;
+            renToken.transfer(sender, _amount);
+            
+            emit RenWithdrawal(sender, _amount);
+
+        }
+        else{
+            // Pool is locked, withdraw will be put in the queue
+            withdrawRequests.push(WithdrawRequest(msg.sender, _amount));           
+        }
 
-        renToken.transfer(sender, _amount);
-        totalPooled -= _amount;
-        balances[sender] -= _amount;
 
-        emit RenWithdrawal(sender, _amount);
     }
 
     function balanceOf(address _addr) external view returns(uint) {
         return balances[_addr];
     }
+
+    function fullfillWithdrawRequest(uint _withdrawId) external payable{
+        // If pool is locked, look if there is a withdraw queue
+        require(isLocked);
+        require(withdrawRequests.length > 0);
+
+        WithdrawRequest memory withdrawRequest = withdrawRequests[_withdrawId];
+
```

Comment:
I think using arrays is potentially dangerous, let's suppose we have the following withdrawRequest array:
```
[{ user: '1', amount: 25 }, { user: '2', amount: 5 }]
```
And now let's imaging 2 users call the fullfillWithdrawRequest method at the same time, the first one tries to deposit 25 and the second one 5. Then, the first call would be `fullfillWithdrawRequest(0)` and the second one `fullfillWithdrawRequest(1)`. Now, what happens if the first request gets proccesed first? Then by the time the second request is processed the `withdrawRequest` has a single item and therefore the index 1 is not reachable and the transaction fails. I think we should use a map or something instead of an array. Maybe look into other projects to see how they do it?

Reply:
Wow indeed! I didn't think about this!
I'll definitely look at other projects to see..